lib/iris/_merge.py

```diff
@@ -1300,7 +1300,11 @@ def _define_space(self, space, positions, indexes, function_matrix):
         def axis_and_name(name):
             axis_dict = {'T': 1, 'Z': 2, 'Y': 3, 'X': 4}
             axis_index = axis_dict.get(self._guess_axis(name), 0)
-            return (axis_index, name)
+            # The middle element ensures sorting is the same as Python 2.
+            if isinstance(name, six.integer_types):
+                return (axis_index, 0, str(name))
+            else:
+                return (axis_index, 1, name)
```

Comment:
Yes, I think so; probably can leave out the `str` to for the same reason as above.